### ISSUE TITLE
Printing PDF attachment

### ISSUE THREAD

**Issue by leonormes** [wavebox/waveboxapp#232]
* Which version of Wavebox are you using? [e.g. 3.0.0]
3.1.8
* Which Operating System are you using? [e.g. Windows 10, Mac OSX 10, Ubuntu 14.04]
MacOSX 10.12.5

I opened an attached PDF and it opened in the preview mode that the web version of Inbox does, but when I pressed the print icon nothing happened. I tried it from Google Inbox web app and it printed fine.

**Comment by Thomas101** [comment]
I think this could be related to #136. I'll take a look thanks

**Comment by Thomas101** [comment]
Hi,

We’ve been working on a fix for this which is due to land in the next release. We’re seeing a couple of crashes from chromium with this release at the moment but on the whole it seems to be reasonably stable. If you want to try it out, it would be great to get some early feedback. I’ve put a early build of Wavebox macOS [here](https://wavebox.io/dl/builds/Wavebox_3_1_8_b07d9c6_osx.dmg). Just a couple of things to bare in mind:
* To install just drag over the previous version of Wavebox.
* To enable the new behaviour
  * Open Settings
  * Click on the Advanced tab
  * Toggle `Use experimental window opener` so it’s enabled
  * Restart Wavebox
* When the next version is released this version will update automatically as usual
* If you want to go back to the currently released version of Wavebox, go to the [Downloads Page](https://wavebox.io/download), download and install over the version you’ve updated

Thanks!

**Comment by b0b069** [comment]
- Which version of Wavebox are you using? [e.g. 3.0.0]
3.1.8 b07d9c6

- Which Operating System are you using? [e.g. Windows 10, Mac OSX 10, Ubuntu 14.04]
MacOSX 10.12.5

Trying out the early build. No dice on printing the PDF straight from gmail. A window pops up, but it's blank, and no print dialogue appears.

The Print All from within gmail is now opening properly in a new window though and allowing the email to be printed (it wasn't before). Personally this was more important for me than printing the PDF, since that can be downloaded and then printed anyways.

**Comment by Thomas101** [comment]
Thanks for the feedback! We'll look into it :)

**Comment by Thomas101** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Print PDF in gmail](https://trello.com/c/kiXybsBu/229-print-pdf-in-gmail)

**Comment by zenyatta80** [comment]
Print doesn't work me either

**Comment by gmonsalvo** [comment]
Hi same issue from here, im using linux. with experimental windows option enabled , when i press the print option a new blank and empty windows opened but with no options.

**Comment by icepol** [comment]
Hi same for me, empty window opened... Version 3.3.0 on Mac...

From console log (from that new window):
```
/Applications/Wavebox.app/Contents/Resources/app.asar/bin/guest/guest/Extensions/extensionLoader.js:91 Uncaught (in promise) Error: Unsupported Guest Protocol
    at ExtensionLoader.loadWaveboxGuestApi (/Applications/Wavebox.app/Contents/Resources/app.asar/bin/guest/guest/Extensions/extensionLoader.js:91:29)
    at new Browser (/Applications/Wavebox.app/Contents/Resources/app.asar/bin/guest/guest/Browser/Browser.js:28:21)
    at GoogleMail.GoogleService (/Applications/Wavebox.app/Contents/Resources/app.asar/bin/guest/guest/Google/GoogleService.js:11:20)
    at GoogleMail (/Applications/Wavebox.app/Contents/Resources/app.asar/bin/guest/guest/Google/GoogleMail.js:21:5)
    at Object.<anonymous> (/Applications/Wavebox.app/Contents/Resources/app.asar/bin/guest/guest/googleMailTooling.js:8:22)
    at Object.<anonymous> (/Applications/Wavebox.app/Contents/Resources/app.asar/bin/guest/guest/googleMailTooling.js:16:3)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
/Applications/Wavebox.app/Contents/Resources/app.asar/bin/guest/guest/Extensions/extensionLoader.js:91 Uncaught (in promise) Error: Unsupported Guest Protocol
    at ExtensionLoader.loadWaveboxGuestApi (/Applications/Wavebox.app/Contents/Resources/app.asar/bin/guest/guest/Extensions/extensionLoader.js:91:29)
    at new NotificationProvider (/Applications/Wavebox.app/Contents/Resources/app.asar/bin/guest/guest/Browser/NotificationProvider.js:22:21)
    at new Browser (/Applications/Wavebox.app/Contents/Resources/app.asar/bin/guest/guest/Browser/Browser.js:32:33)
    at GoogleMail.GoogleService (/Applications/Wavebox.app/Contents/Resources/app.asar/bin/guest/guest/Google/GoogleService.js:11:20)
    at GoogleMail (/Applications/Wavebox.app/Contents/Resources/app.asar/bin/guest/guest/Google/GoogleMail.js:21:5)
    at Object.<anonymous> (/Applications/Wavebox.app/Contents/Resources/app.asar/bin/guest/guest/googleMailTooling.js:8:22)
    at Object.<anonymous> (/Applications/Wavebox.app/Contents/Resources/app.asar/bin/guest/guest/googleMailTooling.js:16:3)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
```

**Comment by Cris70** [comment]
Same here, version 3.3.4 on Linux

**Comment by Thomas101** [comment]
Hi, we’ve just released this to the beta channel of Wavebox in version 3.4.2. If you want to give it a try there are some instructions on [how to use the beta channel here](https://wavebox.io/kb/beta-channel). Thanks

**Comment by gmonsalvo** [comment]
Hi! 
With 3.4.2 Beta, pdf printing works  like a charm! thanks a lot.

**Comment by Thomas101** [comment]
@gmonsalvo awesome news :)

**Comment by Thomas101** [comment]
Hi, this has just been released in the latest version of Wavebox (3.5.0) and Wavebox beta (3.5.1). You can get the update by using the Application/File menu and clicking `Check for Update` or by downloading the latest version from our [download page](https://wavebox.io/download). Thanks!
